cmd/soroban-cli/src/commands/contract/mod.rs

```diff
@@ -174,3 +180,101 @@ pub enum SpecOutput {
 pub struct AliasData {
     ids: HashMap<String, String>,
 }
+
+#[derive(thiserror::Error, Debug)]
+pub enum AliasDataError {
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error("cannot parse contract ID {0}: {1}")]
+    CannotParseContractId(String, DecodeError),
+}
+
+impl AliasData {
+    pub fn load(config_dir: &Path, alias: Option<&str>) -> Result<Option<Self>, AliasDataError> {
+        let Some(alias) = alias else {
+            return Ok(None);
+        };
+
+        let path = Self::alias_path(config_dir, alias);
+
+        if !path.exists() {
+            return Ok(None);
+        }
+
+        let content = fs::read_to_string(path)?;
+        let data: Self = serde_json::from_str(&content).unwrap_or_default();
+
+        Ok(Some(data))
+    }
+
+    pub fn alias_path(config_dir: &Path, alias: &str) -> PathBuf {
+        let file_name = format!("{alias}.json");
+        config_dir.join("contract-ids").join(file_name)
+    }
+
+    pub fn save_contract_id(
+        config_dir: &Path,
+        contract_id: &str,
+        alias: Option<&String>,
+        network_passphrase: &str,
+    ) -> Result<(), AliasDataError> {
+        let Some(alias) = alias else {
+            return Ok(());
+        };
+
+        let path = Self::alias_path(config_dir, alias);
+        let dir = path.parent().ok_or(AliasDataError::CannotAccessConfigDir)?;
+
+        create_dir_all(dir).map_err(|_| AliasDataError::CannotAccessConfigDir)?;
+
+        let content = fs::read_to_string(&path).unwrap_or_default();
+        let mut data: Self = serde_json::from_str(&content).unwrap_or_default();
+
+        let mut to_file = OpenOptions::new()
+            .create(true)
+            .truncate(true)
+            .write(true)
+            .open(path)?;
+
+        data.ids
+            .insert(network_passphrase.into(), contract_id.into());
+
+        let content = serde_json::to_string(&data)?;
+
+        Ok(to_file.write_all(content.as_bytes())?)
+    }
+
+    pub fn get_contract_id(
+        alias: &str,
+        config_dir: &Path,
+        network_passphrase: &str,
+    ) -> Result<Option<String>, AliasDataError> {
+        let alias_data = Self::load(config_dir, Some(alias))?;
+        let Some(alias_data) = alias_data else {
+            return Ok(None);
+        };
+
+        match alias_data.ids.get(network_passphrase) {
+            Some(id) => Ok(Some(id.into())),
+            _ => Ok(None),
+        }
+    }
+
+    pub fn load_contract_id_or_default(
+        alias: &str,
```

Comment:
It is a little confusing that this could also be a contract strkey. So I suggest this just return an `Option` of a contract Strkey. But for now I'd just change the parameter name to make it clearer.

Reply:
Can you please explain why returning an `Option` be more clearer? The idea is avoiding this common task on the caller, as defaulting to the input address when no alias can be found will always be the desirable outcome.

Reply:
Personally I think it's a clearer API if the alias tries to resolve the alias and it's None if not there.

Then you could have a method on the config `Arg` that would handle this higher level. TL;DR keep this part simple and then the callers in `invoke` etc can do `config.resolve_contract_id(&self.contract_id)` and it will handle the last part of this method.